test/gst-vaapi/transcode/transcoder.py

```diff
@@ -11,44 +11,177 @@
+    return self.requirements[ttype].get(
+      codec, {}).get(mode, ([], (False, False), None))
 
-  def transcode_1to1(self):
-    self.transcoded = get_media()._test_artifact(
-      "{case}_{width}x{height}_{mode}.{dstextension}".format(**vars(self)))
-    get_media().test_call_timeout = vars(self).get("call_timeout", 0)
-    call("gst-launch-1.0 -vf filesrc location={source} ! {gsttrans}"
-      " ! filesink location={transcoded}".format(**vars(self)))
+  def validate_spec(self):
+    from slash.utils.pattern_matching import Matcher
+
+    assert len(self.outputs), "Invalid test case specification, outputs data empty"
+    if len(self.outputs) > 1 or self.outputs[0].get("channels", 1) > 1:
+      slash.skip_test("transcode 1ton tests not supported")
+    assert self.mode in ["sw", "hw"], "Invalid test case specification as mode type not valid"
+
+    # generate platform list based on test runtime parameters
+    iplats, ireq, _ =  self.get_requirements_data("decode", self.codec, self.mode)
+    platforms = set(iplats)
+    requires = [ireq,]
+
+    codec = self.outputs[0]["codec"]
+    mode  = self.outputs[0]["mode"]
+    assert mode in ["sw", "hw"], "Invalid test case specification as output mode type not valid"
+    oplats, oreq, _ = self.get_requirements_data("encode", codec, mode)
+    platforms &= set(oplats)
+    requires.append(oreq)
+
+    # create matchers based on command-line filters
+    matchers = [Matcher(s) for s in slash.config.root.run.filter_strings]
+
+    # check if user supplied any platform tag on command line
+    pmatch = any(map(lambda p: any([m.matches(p) for m in matchers]), ALL_PLATFORMS))
+
+    # if user supplied a platform tag, check if this test can support it via the
+    # test param-based required platforms
+    if pmatch and not any(map(lambda p: any([m.matches(p) for m in matchers]), platforms)):
+      slash.skip_test("unsupported platform")
+
+    # check required
+    if not all([t for t,m in requires]):
+      slash.skip_test("One or more software requirements not met")
 
-    self.convert_toyuv()
-    self.check_metrics()
+  def gen_input_opts(self):
+    opts = " -vf filesrc location={source}"
+    _, _, ffdecparms = self.get_requirements_data("decode", self.codec, self.mode)
+    assert ffdecparms is not None, "gen_input_opts decode parameters are empty"
+    ffdecoder, _ = ffdecparms
+    opts += " ! {}".format(ffdecoder)
+
+    return opts.format(**vars(self))
+
+  def gen_output_opts(self):
+    opts = ""
+    codec = self.outputs[0]["codec"]
+    mode  = self.outputs[0]["mode"]
+    _, _, ffencparms = self.get_requirements_data("encode", codec, mode)
+    assert ffencparms is not None, "gen_output_opts encode parameters are empty"
+    ffencoder, ext = ffencparms
+    opts += " ! {}".format(ffencoder)
+    self.ofile = get_media()._test_artifact(
+      "{}.{}".format(self.case, ext))
+    opts += " ! filesink location={}".format(self.ofile)
```

Comment:
Since you already call `format` upon method return, you can avoid the extra call to `format` here like so:
```
opts += " ! filesink location={ofile}"
```

Reply:
modified it